### PR TITLE
Validate parsing invalid usage of action

### LINKED ISSUE #29430: Semantic Analyzer fails with CCE
**Description:**
Consider the following source snippet where we do not have quotes around the URI

```
function caller() {
    ws:Client wsClient = checkpanic  new(wss://ws.finnhub.io/?token);

   checkpanic  wsClient->writeTextMessage("{\"type\":\"subscribe\",\"symbol\":\"AAPL\"}");

   string textResp = checkpanic  wsClient->readTextMessage();
   io:println(textResp);
}
```

The throws the following CCE

```
ballerinalang.compiler.tree.expressions.BLangCheckPanickedExpr cannot be cast to class org.wso2.ballerinalang.compiler.tree.expressions.BLangAccessExpression (org.wso2.ballerinalang.compiler.tree.expressions.BLangCheckPanickedExpr and org.wso2.ballerinalang.compiler.tree.expressions.BLangAccessExpression are in unnamed module of loader 'app')'}
java.lang.ClassCastException: class org.wso2.ballerinalang.compiler.tree.expressions.BLangCheckPanickedExpr cannot be cast to class org.wso2.ballerinalang.compiler.tree.expressions.BLangAccessExpression (org.wso2.ballerinalang.compiler.tree.expressions.BLangCheckPanickedExpr and org.wso2.ballerinalang.compiler.tree.expressions.BLangAccessExpression are in unnamed module of loader 'app')
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkActionInvocation(TypeChecker.java:5321)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:2808)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangInvocation$BLangActionInvocation.accept(BLangInvocation.java:222)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:370)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:349)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkTypeParamExpr(TypeChecker.java:5834)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkTypeParamExpr(TypeChecker.java:5827)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkInvocationArgs(TypeChecker.java:5546)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkInvocationParam(TypeChecker.java:5496)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:3208)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangTypeInit.accept(BLangTypeInit.java:62)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:370)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:349)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visitCheckAndCheckPanicExpr(TypeChecker.java:4815)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:4438)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangCheckPanickedExpr.accept(BLangCheckPanickedExpr.java:44)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:370)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:349)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:846)
```

**Affected Versions:**
sl-alpha3 at least

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes #29430

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by rdulmina** [comment]
@IrushiL can u verify the formatter changes in 59fe8f8

**Review by lochana-chathura — COMMENTED** [review]
Parser changes look gd!

File: misc/formatter/modules/formatter-core/src/test/resources/actions/start/assert/start_action_2.bal [hunk 1]
```diff
@@ -1 +1,3 @@
-future<int> f1 = @strand {thread: "any"} start multiply(1, 2);
+function foo() {
+    future<int> f1 = @strand {thread: "any"} start multiply(1, 2);
+}
```
Comment (lochana-chathura) [inline misc/formatter/modules/formatter-core/src/test/resources/actions/start/assert/start_action_2.bal:3]: formattar changes also LGTM. as `future<int> f1 = @strand {thread: "any"} start multiply(1, 2);` is negative scenario at top level.

Comment (rdulmina) [reply misc/formatter/modules/formatter-core/src/test/resources/actions/start/assert/start_action_2.bal:3]: Thanx @lochana-chathura

**Review by lochana-chathura — APPROVED** [review]

**Pull request merged**